R/utils.R

```diff
@@ -1046,3 +1046,23 @@ URLencode <- function(value, reserved = FALSE) {
   value <- enc2utf8(value)
   if (reserved) encodeURIComponent(value) else encodeURI(value)
 }
+
+
+# This function takes a name and function, and it wraps that function in a new
+# function which calls the original function using the specified name. This can
+# be helpful for profiling, because the specified name will show up on the stack
+# trace.
+wrapFunctionLabel <- function(func, name) {
+  if (name == "name" || name == "func" || name == "relabelWrapper") {
+    stop("Invalid name for wrapFunctionLabel: ", name)
+  }
+  assign(name, func, environment())
+
+  relabelWrapper <- function(...) {
+    # This `f` gets renamed to the value of `name`
+    f(...)
+  }
+
+  body(relabelWrapper)[[2]][[1]] <- as.name(name)
```

Comment:
I just tested that out, and it has a somewhat weird result. When printed, the function show something different from the actual body of the function, because of source refs stored in the function.

```R
x <- eval(substitute(
  function(...) {
    f(...)
  },
  list(f = as.name("FOO"))
))

# When printed, it says it's calling f
x
# function(...) {
#     f(...)
#   }

# But it's really calling FOO
body(x)
# {
#     FOO(...)
# }

str(x)
# function (...)  
#  - attr(*, "srcref")=Class 'srcref'  atomic [1:8] 2 3 4 3 3 3 2 4
#   .. ..- attr(*, "srcfile")=Classes 'srcfilecopy', 'srcfile' <environment: 0x93d0018> 
```

But with my method, it prints the actual  content of the function:

```R
z <- function(...) {
  f(...)
}
body(z)[[2]][[1]] <- as.name("FOO")

z
# function (...) 
# {
#     FOO(...)
# }

body(z)
# {
#     FOO(...)
# }

# No source refs
str(z)
# function (...)  
```

But other than that, I think using `substitute` is a cleaner approach, so I'll go with that.